core/metacling/src/TCling.cxx

```diff
@@ -1463,23 +1500,23 @@ static std::string FindLibraryName(void (*func)())
       char buf[PATH_MAX] = { 0 };
       uint32_t bufsize = sizeof(buf);
       if (_NSGetExecutablePath(buf, &bufsize) >= 0)
-         return buf;
-      return info.dli_fname;
+         return ResolveSymlink(buf);
+      return ResolveSymlink(info.dli_fname);
```

Comment:
@vgvassilev , I think you also need to resolve the symlink at https://github.com/vgvassilev/root/blob/481bf12364279089880d49dc06794fedeb9602a0/core/metacling/src/TCling.cxx#L1495

Reply:
Yep, indeed that was one of the issues, thanks! I needed a bunch of other changes too...

Reply:
I just checked this locally and we still have the same issue. Could this be related to https://github.com/root-project/root/blob/481bf12364279089880d49dc06794fedeb9602a0/core/metacling/src/TCling.cxx#L1800

```
>root.exe -l -q -b
Error in <TCling::LoadPCM>: ROOT PCM /build/muz/x/w/slc7_amd64_gcc700/lcg/root/6.17.01-muz/lib/libMathCore_rdict.pcm file does not exist
Error in <TCling::LoadPCM>: ROOT PCM /build/muz/x/w/slc7_amd64_gcc700/lcg/root/6.17.01-muz/lib/libNet_rdict.pcm file does not exist
Error in <TCling::LoadPCM>: ROOT PCM /build/muz/x/w/slc7_amd64_gcc700/lcg/root/6.17.01-muz/lib/libTree_rdict.pcm file does not exist
Error in <TCling::LoadPCM>: ROOT PCM /build/muz/x/w/slc7_amd64_gcc700/lcg/root/6.17.01-muz/lib/libMatrix_rdict.pcm file does not exist
Error in <TCling::LoadPCM>: ROOT PCM /build/muz/x/w/slc7_amd64_gcc700/lcg/root/6.17.01-muz/lib/libHist_rdict.pcm file does not exist

```